### ISSUE TITLE
Add WITH(NOLOCK) clause in SELECT SQL query

### ISSUE THREAD

**Issue by mukarramjavid** [borisdj/EFCore.BulkExtensions#1305]
Hi, 
Is there any plan to add the **WITH(NOLOCK)** clause in the SELECT statement within the BulkRead() method?
For example,
`SELECT *
FROM Employees WITH(LOCK)`

Thanks

**Comment by LeDahu22** [comment]
@mukarramjavid  Maybe with (readpast) would be preferable, otherwise you could read uncommited dirty data.

**Comment by mukarramjavid** [comment]
> @mukarramjavid Maybe with (readpast) would be preferable, otherwise you could read uncommited dirty data.

@borisdj Yes, it is preferable. However, if you make it optional, the user can choose whether to add `NO(LOCK)` / `READPAST` or not, similar to `WITH(HOLDLOCK)`. What do you think?
